src/main/java/com/celements/course/service/CourseService.java

```diff
@@ -315,6 +315,39 @@ public boolean validateParticipant(DocumentReference regDocRef, String emailAdr,
     return success;
   }
 
+  @Override
+  public CourseConfirmState getConfirmState(DocumentReference regDocRef) {
+    DocumentReference courseParticipantClassRef = getCourseClasses().getCourseParticipantClassRef(
+        modelContext.getXWikiContext().getDatabase());
+    CourseConfirmState confirmState = CourseConfirmState.UNDEFINED;
+    try {
+      List<BaseObject> partiObjs = modelAccess.getXObjects(regDocRef, courseParticipantClassRef);
```

Comment:
also, leads to NPE if regDocRef is null, yet this is possible according to current contract. change to NotNull